Debugging and adding a failing test.

THOR300 opened the pull request:
# Description

This test is just to make it super clear that we are not getting `geographies` back from vespa (not necessarily looking to merge). 


<img width="963" alt="image" src="https://github.com/user-attachments/assets/283e9708-da62-4155-b66b-1fb039cf3113">



## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [x] Skip auto-tagging
- [] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail (integrated soon)
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [ ] Bug fix
- [ ] New feature
- [ ] Breaking change

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Before submitting

<!-- Please complete this checklist BEFORE submitting your PR to speed along the review process. -->
- [ ] I've read and followed all steps in the [Making a pull request](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#making-a-pull-request)
    section of the `CONTRIBUTING` docs.
- [ ] I've updated or added any relevant docstrings following the syntax described in the
    [Writing docstrings](https://github.com/climatepolicyradar/cpr-sdk/blob/main/.github/CONTRIBUTING.md#writing-docstrings) section of the `CONTRIBUTING` docs.
- [ ] If this PR fixes a bug, I've added a test that will fail without my fix.
- [ ] If this PR adds a new feature, I've added tests that sufficiently cover my new functionality.

olaughter commented on src/cpr_sdk/search_adaptors.py:
```diff
@@ -90,6 +90,8 @@ def search(self, parameters: SearchParameters) -> SearchResponse:
                 raise e
         query_time_end = time.time()
 
+        # TODO: Confirmed that here we have geography but no geographies in the
```
Items should also be on the passage that are not on the passage so we can return that context - but that doesn't explain why vespa is returning nothing. I think we can fix this one with a schema change and redeploy

THOR300 commented:
Bug

No geographies in search response on staging when using “exact match, region, query” 

We have test coverage for this so how did this get through? 
- I’ve added a failing test that mimics this permutation of search filters on the app. 
    - https://github.com/climatepolicyradar/cpr-sdk/pull/108
- The issue comes from the query returning only passage hits and not document hits. Passages seem to be coming back with no geographies field.
    - Why this occurs I’m not sure. 
- I’ve validated locally by stepping through the code that we are getting geographies back for the relevant tests in the backend and cpr_sdk.
    - Geographies is present for these as we do get document hits. 

So why does this failing test get no document hits whereas others do? 
- I’m searching for ‘e’ in the test that fails to return document type responses, that text definitely exists in the titles of the documents? 

No document type hits = No geographies in response


CPR_SDK
Calling staging: 
- [ ] Confirmed that on staging we are getting no hits with geographies present in the vespa_response
    - [ ] This means that when we try and create Passage type hits there’s no geographies present. 
    - [ ] How? By using the sdk with the correct certs and putting in a breakpoint in the code before we post process
- [ ] Confirmed that we are only getting passage and not document hits when calling the staging instance. 

Tests:
- [ ] Confirmed that there are no document type hits and no geographies in the response using this test: 
    - [ ] tests/test_search_adaptors.py::test_vespa_search_response__geographies
- [ ] Confirmed that there are document type hits and geographies present in this test: 
    - [ ] tests/test_search_adaptors.py::test_vespa_search_adaptor__filters

BACKEND
- [ ] Confirmed that on main in the backend we are getting geographies in results in tests and this is why they are passing. 
    - [ ] Using this test and adding a breakpoint: tests/search/vespa/test_vespa_search.py::test_metadata_filter


Potential Fixes: 

Two options: 
1. Get geographies back for all query behaviour

This is confusing as it should be a join on the family! 
- Adding geographies to the document_passage summary does not fix this either! 

2. Try and understand why some query behaviour only returns passages and change it to always return geographies.

olaughter commented on tests/test_search_adaptors.py:
```diff
@@ -553,3 +556,39 @@ def test_vespa_search_adaptor__filters(test_vespa, query_string, filters):
+    vespa_response = test_vespa.client.query(body=build_vespa_request_body(parameters))
+
+    root = vespa_response.json["root"]
+    response_families = dig(root, "children", 0, "children", 0, "children", default=[])
+    for family in response_families:
+        for hit in dig(family, "children", 0, "children", default=[]):
+            hit = Hit.from_vespa_response(response_hit=hit)
+            assert hit.family_geography not in [None, []]
+            assert hit.family_geographies not in [None, []]
```
This test checks that the fields are returned, effectively testing the summary part of the schema (as well as sdk handling). Seems the issue is prior to that in the actual object itself needing the fields to scan on the passage object

olaughter committed:
Update schema to add new fields to passage object

We spotted issues inf iltering that have turned out to originate from
these being missing. Essentially the yql was skipping searching passages
because these fields where not there. So in the case of an exact match
query, we give it an impossible task of searching for text blocks, which
are passage specific AND for geographies which are document specific.

The issue seems to have been there on other queries but was masked by
the way the other queries balance other items being searched.

olaughter committed:
Prevent false positive courpus fields test

This returned no hits previously. We missed it because the for loop never
got to any asserts.

olaughter committed:
Add corpus items to passages in post processing

olaughter committed:
Ensure geography fields are returned

jesse-c reviewed (approved):
LGTM

Took me a second to appreciate the `asserts` in the test. 😅
